backend/app.ts

```diff
@@ -0,0 +1,76 @@
+import * as dotenv from 'dotenv';
+import path from 'path';
+dotenv.config({ path: path.resolve(__dirname, "../.env") });
+import express from 'express';
+import cors from 'cors';
+import morgan from 'morgan';
+import session from 'express-session';
+import passport from 'passport';
+const GoogleStrategy = require('passport-google-oauth20').Strategy;
+import googleOAuthRouter from './routes/google-oauth';
+import testRouter from './routes/test-route'
+import prisma from './utils/prismaClient';
+
+const app = express();
+
+dotenv.config();
+app.use(morgan("dev"));
+app.use(cors());
+app.use(express.json());
+app.use(session({
+    secret: "secretcode",
+    resave: true,
+    saveUninitialized: true,
```

Comment:
genuine question, maybe good for @chinkevin25 to chime in on too: should these options both be set to true?

from docs: https://www.npmjs.com/package/express-session

resave:

> How do I know if this is necessary for my store? The best way to know is to check with your store if it implements the touch method. If it does, then you can safely set resave: false. If it does not implement the touch method and your store sets an expiration date on stored sessions, then you likely need resave: true.

saveUninitialized:
> Forces a session that is "uninitialized" to be saved to the store. A session is uninitialized when it is new but not modified. Choosing false is useful for implementing login sessions, reducing server storage usage, or complying with laws that require permission before setting a cookie. Choosing false will also help with race conditions where a client makes multiple parallel requests without a session.

Reply:
@chinkevin25 can you take a look at two options? The way I interpreted **resave** is that when a user visits a page, their session will saved even when the session was not modified. Always keeping the user session on our database.

For ** saveUninitialized**, as soon the user visits the website, a session is created for them. I think this should be true (?) unless we only create sessions for people with accounts.

Reply:
For `saveUninitialized`, like you mentioned it'll save a session to the session store immediately. Like you mentioned it'll be true if we needed to store information on the session other than login. We might include this if we were doing more tracking. For now, however, we're only going to be doing login sessions. So we could opt for false, if we wanted to. Either `true` or `false` works for our purpose, so it's just best to understand when you'd want it on.

Another reason we might want immediate session saving, is if we needed to carry information from a logged out state to a logged in state, or if we wanted to see if "repeat" clients came to visit. I'd lean to leaving it `false` until we have a need to turn it on just to save DB space.

Reply:
For resave, since you are not setting a `store` value, I believe you're using the default MemoryStorage which I don't think has `touch` implemented. So you should have resave `true` as you have it.

It's worth mentioning that `MemoryStore` will need to be replaced in production, but can be done later.